From 7b01a3d451734b9db47a7818ade3befc2fcb6a12 Mon Sep 17 00:00:00 2001
From: Phil Elwell <phil@raspberrypi.com>
Date: Thu, 16 Nov 2023 16:03:47 +0000
Subject: [PATCH 0824/1085] ARM: dts: bcm2712: Prune the non-D0 hardware

There is no point describing hardware blocks in C0/1 that aren't also
in D0, so delete them.

Signed-off-by: Phil Elwell <phil@raspberrypi.com>
---
 .../arm/boot/dts/broadcom/bcm2712-rpi-5-b.dts |  2 -
 arch/arm/boot/dts/broadcom/bcm2712.dtsi       | 66 -------------------
 2 files changed, 68 deletions(-)

--- a/arch/arm/boot/dts/broadcom/bcm2712-rpi-5-b.dts
+++ b/arch/arm/boot/dts/broadcom/bcm2712-rpi-5-b.dts
@@ -24,8 +24,6 @@
 #define spi6 _spi6
 #define uart0 _uart0
 #define uart2 _uart2
-#define uart3 _uart3
-#define uart4 _uart4
 #define uart5 _uart5
 
 #include "bcm2712.dtsi"
--- a/arch/arm/boot/dts/broadcom/bcm2712.dtsi
+++ b/arch/arm/boot/dts/broadcom/bcm2712.dtsi
@@ -197,28 +197,6 @@
 			status = "disabled";
 		};
 
-		uart3: serial@7d001600 {
-			compatible = "arm,pl011", "arm,primecell";
-			reg = <0x7d001600 0x200>;
-			interrupts = <GIC_SPI 121 IRQ_TYPE_LEVEL_HIGH>;
-			clocks = <&clk_uart>,
-				 <&clk_vpu>;
-			clock-names = "uartclk", "apb_pclk";
-			arm,primecell-periphid = <0x00241011>;
-			status = "disabled";
-		};
-
-		uart4: serial@7d001800 {
-			compatible = "arm,pl011", "arm,primecell";
-			reg = <0x7d001800 0x200>;
-			interrupts = <GIC_SPI 121 IRQ_TYPE_LEVEL_HIGH>;
-			clocks = <&clk_uart>,
-				 <&clk_vpu>;
-			clock-names = "uartclk", "apb_pclk";
-			arm,primecell-periphid = <0x00241011>;
-			status = "disabled";
-		};
-
 		uart5: serial@7d001a00 {
 			compatible = "arm,pl011", "arm,primecell";
 			reg = <0x7d001a00 0x200>;
@@ -543,17 +521,6 @@
 			status = "disabled";
 		};
 
-		uartc: serial@7d50e000 {
-			compatible = "brcm,bcm7271-uart";
-			reg = <0x7d50e000 0x20>;
-			reg-names = "uart";
-			reg-shift = <2>;
-			reg-io-width = <4>;
-			interrupts = <GIC_SPI 278 IRQ_TYPE_LEVEL_HIGH>;
-			skip-init;
-			status = "disabled";
-		};
-
 		aon_intr: interrupt-controller@7d510600 {
 			compatible = "brcm,bcm2711-l2-intc", "brcm,l2-intc";
 			reg = <0x7d510600 0x30>;
@@ -1103,30 +1070,6 @@
 			brcm,msi-pci-addr = <0xff 0xffffe000>;
 		};
 
-		genet: ethernet@1300000 {
-			compatible = "brcm,bcm2711-genet-v5";
-			reg = <0x10 0x01300000  0x0 0x20010>;
-			#address-cells = <0x1>;
-			#size-cells = <0x0>;
-			interrupts = <GIC_SPI 265 IRQ_TYPE_LEVEL_HIGH>,
-				     <GIC_SPI 266 IRQ_TYPE_LEVEL_HIGH>;
-			status = "disabled";
-			phy-mode = "rgmii";
-			fixed-link = <0x0 0x1 0x3e8 0x0 0x0>;
-	                  phy-speed = <0x3e8>;
-	                  phy-id = <0x101>;
-	                  phy-type = <0x6>;
-	                  local-mac-address = [ 00 10 18 d8 45 de ];
-	                  device_type = "network";
-
-			genet_mdio: mdio@e14 {
-				compatible = "brcm,genet-mdio-v5";
-				reg = <0xe14 0x8>;
-				#address-cells = <0x1>;
-				#size-cells = <0x0>;
-			};
-		};
-
 		syscon_piarbctl: syscon@400018 {
 			compatible = "brcm,syscon-piarbctl", "syscon", "simple-mfd";
 			reg = <0x10 0x00400018  0x0 0x18>;
@@ -1188,15 +1131,6 @@
 			mmc-ddr-3_3v;
 			status = "disabled";
 		};
-
-		sdio0: mmc@1108000 {
-			compatible = "brcm,bcm2711-emmc2";
-			reg = <0x10 0x01108000  0x0 0x100>;
-			interrupts = <GIC_SPI 272 IRQ_TYPE_LEVEL_HIGH>;
-			clocks = <&clk_emmc2>;
-			mmc-ddr-3_3v;
-			status = "disabled";
-		};
 
 		bcm_reset: reset-controller@1504318 {
 			compatible = "brcm,brcmstb-reset";
